display/ssd1306.h

```diff
@@ -252,14 +278,17 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
   }
 
   void ShowFile(Effect* effect) {
-    MountSDCard();
-    eof_ = true;
-    file_.Play(effect);
-    frame_available_ = false;
-    loop_start_ = millis();
-    frame_count_ = 0;
-    SetScreenNow(SCREEN_IMAGE);
-    eof_ = false;
+    if (*effect) {
+      MountSDCard();
+      eof_ = true;
+      file_.Play(effect);
+      frame_available_ = false;
+      loop_start_ = millis();
+      frame_count_ = 0;
+      SetScreenNow(SCREEN_IMAGE);
+      eof_ = false;
+      on_played_ = effect == &IMG_on;
```

Comment:
on_played_ sounds like "on" has played at any time in the past.
In fact, I think it might be better to just have a pointer to the effect and call it current_effect_.